[CUBRIDMAN-275] [EN] 시스템 패키지: DBMS_OUTPUT 설명 추가 (#625)
CUBRID/cubrid-manual: en/pl/pl_package.rst

-----------------------------
System Package
-----------------------------

CUBRID provides system packages for user convenience. 
More packages will be added in future versions to expand CUBRID's functionality, but currently, only the DBMS_OUTPUT package is available.

.. _dbms_output:

DBMS_OUTPUT
==============================

The DBMS_OUTPUT package provides functionality to store and read string messages in the DBMS_OUTPUT buffer. 

> Developers of stored procedures/functions can use the PUT_LINE or PUT functions of this package to accumulate desired messages in the DBMS_OUTPUT buffer.

Reviewer: ```suggestion
Developers of stored procedures/functions can use the PUT_LINE or PUT functions of this package to accumulate relevant messages in the DBMS_OUTPUT buffer.
```

Client tools like CSQL or DBeaver use the ENABLE, DISABLE, GET_LINE, and GET_LINES functions of this package 
to activate/deactivate the message storage feature and retrieve messages accumulated in the buffer. 

> Developers can use these messages effectively for checking program progress or debugging.

Reviewer: ```suggestion
Developers can effectively use these messages for monitoring program progress or debugging.
```

This section explains the usage and examples of the DBMS_OUTPUT package.
The functions of the DBMS_OUTPUT package are as follows:

        * :ref:`dbms-output-enable`
        * :ref:`dbms-output-disable`
        * :ref:`dbms-output-put`
        * :ref:`dbms-output-put_line`
        * :ref:`dbms-output-new_line`
        * :ref:`dbms-output-get_line`
        * :ref:`dbms-output-get_lines`

.. _dbms-output-enable:

DBMS_OUTPUT.ENABLE
----------------------

.. function:: DBMS_OUTPUT.ENABLE (size)

        Enable the DBMS_OUTPUT message buffer and sets the size of the buffer to store messages.

        :param size: Specifies the size of the buffer in bytes. The maximum size is 32767 bytes, and exceeding this value will result in an error.

.. note::

        Calling **;server-output on** in the CSQL interpreter is internally equivalent to calling the default DBMS_OUTPUT.ENABLE(20000).
        For more details, refer to :ref:`CSQL session command server-output <server-output>`.

.. _dbms-output-disable:

DBMS_OUTPUT.DISABLE
----------------------

.. function:: DBMS_OUTPUT.DISABLE ()

>         Removes the messages stored in the current buffer and deactivates the buffer.

Reviewer: ```suggestion
        Clears the messages stored in the current buffer and deactivates it.
```

>         Therefore, no output will appear even if other procedures in the DBMS_OUTPUT package are called.

Reviewer: ```suggestion
        Consequently, no output will appear even if other procedures in the DBMS_OUTPUT package are called.
```

        For more details, refer to :ref:`CSQL session command server-output <server-output>`.

.. _dbms-output-put:

DBMS_OUTPUT.PUT
----------------------

.. function:: DBMS_OUTPUT.PUT (str VARCHAR)

        Stores the specified string in the buffer without a newline.

        :param str: Specifies the string to store. If the string to store is NULL, no action is taken.

.. _dbms-output-put_line:

DBMS_OUTPUT.PUT_LINE
----------------------

.. function:: DBMS_OUTPUT.PUT_LINE (line VARCHAR)

        Stores the specified string in the buffer and adds a newline.

        :param line: Specifies the string to store. If the string to store is NULL, no action is taken.

.. _dbms-output-new_line:

DBMS_OUTPUT.NEW_LINE
----------------------

.. function:: DBMS_OUTPUT.NEW_LINE ()

        Adds a newline character to the buffer. After adding a string with the PUT function, call the NEW_LINE function to read it line by line with GET_LINE.

.. _dbms-output-get_line:

DBMS_OUTPUT.GET_LINE
----------------------

.. function:: DBMS_OUTPUT.GET_LINE (line OUT VARCHAR, status OUT INTEGER)

>         Reads the first line of the string message stored in the buffer. The read line is deleted from the buffer.

Reviewer: ```suggestion
        Reads the first line of the string message stored in the buffer. The read line is then deleted from the buffer.
```
